tests/conftest.py

```diff
@@ -177,25 +180,37 @@ def softfork_height(request) -> int:
 
 @pytest.fixture(scope="session")
 def default_400_blocks(bt):
+    version = ""
+    if bt.constants.SOFT_FORK3_HEIGHT == 0:
```

Comment:
I think it would be clearer to depend on the `consensus_mode` fixture here, and check if you're in `Mode.SOFT_FORK3`. There are a few similar cases below as well.